### PR TITLE
feat: Adds Orbit Community Search Widget

### PR DESCRIPTION
**Pull request by dthyresson**
This PR adds the Orbit Community Search for RedwoodJS (we're beta-ing):

image

image

More info on Community Search:

* https://search.orbit.love/redwoodjs/setup
* https://orbit.love/docs/all/community-search

Known issues:

* Somertimes `Blocked a frame with origin "http://localhost:4318/" from accessing a frame with origin "https://search.orbit.love/".  The frame requesting access has a protocol of "http", the frame being accessed has a protocol of "https". Protocols must match.` because dev server is http
* Am asking Orbit about ways to mitigate

### CODE REVIEW

**Comment by dthyresson** [comment]
Some updates:

1. Move search to sidebar as quick search for community.
2. Search has analytics now. This is within the Orbit platform: 
image

**Comment by Tobbe** [comment]
How do I run this locally? When I try to start studio now I get this in my console
image

**Comment by dthyresson** [comment]
> How do I run this locally? When I try to start studio now I get this in my console image
> 
> And no search box
> 
> image

This was one of the open issues, and I realized I should be initing the widget in the Sidebar .... and apparently the way to do this is to have useEffect create the script tab dynamically on mounting and remove when unmouting.

I think the widget inits more reliably now.

File: web/src/layouts/SidebarLayout/SidebarLayout.tsx [hunk 1]
```diff
@@ -13,6 +13,42 @@ type SidebarLayoutProps = {
...
+  `
+    document.body.appendChild(script)
+    document.body.appendChild(script)
+
+    return () => {
+      document.body.removeChild(script)
+    }
+  }, [])
+
+  useEffect(() => {
```
Comment (Tobbe) [inline web/src/layouts/SidebarLayout/SidebarLayout.tsx:40]: This looks a little bit weird. Does orbit have any docs about how to best integrate this into a React page?

File: web/src/layouts/SidebarLayout/SidebarLayout.tsx [hunk 2]
```diff
@@ -13,6 +13,42 @@ type SidebarLayoutProps = {
...
+        searchBoxPlaceholder: "Community search...",
+      },
+    });
+
+    if (window.CommunitySearch.init) {
+      window.CommunitySearch.init();
+    }
+  `
+    document.body.appendChild(script)
+    document.body.appendChild(script)
```
Comment (Tobbe) [inline web/src/layouts/SidebarLayout/SidebarLayout.tsx:33]: You don't want this line duplicated, do you?

Comment (dthyresson) [reply web/src/layouts/SidebarLayout/SidebarLayout.tsx:33]: That's what happens when you commit 30 seconds before a DP call :)

**Comment by Tobbe** [comment]
I think we should get rid of these
image

I don't think anyone will open Studio to find the links to our Discord and Forums :) They'd just go straight there instead. Plus it makes the menu scroll, which looks horrible.

**Comment by Tobbe** [comment]
I've updated the code that loads the search widget. But you forgot to add the widget to the mobile menu
image

And I'm afraid it's not entirely straight forward to include it there as well. Would have to re-initialize the widget with a new target div #id and I'm not sure what Orbit's API for that is. Plus we need to tell React to unload/reload relevant code.

Getting docs written for what we have is now more important than adding more features like this.

**Comment by dthyresson** [comment]
Another option is to make our own search input and toggle the community search opening. This won’t use the container option but pure manual.

**Comment by Tobbe** [comment]
Looking at the code some more I realized this is already producing invalid html 🙁 

The `<NavigationMenu />` component is inserted in two places (once for desktop and once for mobile), which means the `#community-search-box` id will show up twice in the DOM, and `id`s aren't allowed to do that. They need to be unique.

**Comment by Tobbe** [comment]
> Another option is to make our own search input and toggle the community search opening. This won’t use the container option but pure manual.

I took this approach with my latest commit. It solves both the invalid html issue I noted + we get the search button in both menus

@dthyresson If you decide to work on this today there are two outstanding issues with the search:

 * I just copy/pasted the orbit source code. We need to switch to Tailwind instead of inline styles
 * We need to implement support for dark/light mode. It's hardcoded to darkmode right now

**Pull request merged**